File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/handler/SLEventQueueMessageHandler.java
package com.lumiyaviewer.lumiya.slproto.handler;

import com.lumiyaviewer.lumiya.slproto.caps.SLCapEventQueue;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface SLEventQueueMessageHandler {
    SLCapEventQueue.CapsEventType eventName();
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/handler/SLEventQueueMessageHandler.kt
package com.lumiyaviewer.lumiya.slproto.handler

import com.lumiyaviewer.lumiya.slproto.caps.SLCapEventQueue

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class SLEventQueueMessageHandler(val eventName: SLCapEventQueue.CapsEventType)
